thrust/thrust/system/tbb/detail/reduce_by_key.inl

```diff
@@ -63,18 +64,23 @@ inline L divide_ri(const L x, const R y)
 template <typename InputIterator, typename BinaryFunction, typename SFINAE = void>
 struct partial_sum_type
 {
-  using type = typename thrust::iterator_value<InputIterator>::type;
+  using value_t = thrust::iterator_value_t<InputIterator>;
+  // TODO(bgruber): unify with CUB's accumulator_t
+  using type = ::cuda::std::__decay_t<typename ::cuda::std::__invoke_of<BinaryFunction, value_t, value_t>::type>;
```

Comment:
important: following the same logic as below, let's have uniform behavior across execution policies. We'll revisit transition to invoke result after porting CUDA backend to use CUB. 
```suggestion
  using type = thrust::iterator_value_t<InputIterator>;
```